### ISSUE TITLE
Request chocolatey (choco) package for CloudCompare

### ISSUE THREAD

**Issue by jo-chemla** [CloudCompare/CloudCompare#1728]
**Describe the feature you would like**
[Chocolatey](https://chocolatey.org/) is one of the most used package managers for windows - along with the more recent [winget](https://winget.run/) alternative from msft. It would be good for deployment to have a chocolatey package for CloudCompare. This way, individuals - or orgs managing lots of machines - would easily deploy a given version of CloudCompare and stay up-to-date.

Here are some thoughts I posted on https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/issues/903 to build a chocolatey package. 
 - Since you are using Github Actions, [winget-releaser](https://github.com/marketplace/actions/winget-releaser) is a Github Action made for winget automatic package deployment during CI. A very useful [resource](https://www.dolthub.com/blog/2021-10-13-how-to-publish-winget-chocolatey-packages-with-github-actions/) explains in details how to use Github Actions YAML syntax to push choco and winget packages.
 - Otherwise, we could also try to publish a Request for Package (RFP) on [chocolatey-community/chocolatey-package-requests](https://github.com/chocolatey-community/chocolatey-package-requests) so that another maintainer would do these choco releases every time there is a github release for SSHFS-win-manager.

Hope this helps, and as always, thanks a lot for this exceptional piece of software!

**Describe alternatives you've considered**
Alternative: nuget winget package, or an ansible script to download CloudCompare from the official website, unzip, add to path or run the installer. 

**Is your feature request related to a problem? Please describe.**
Not a problem, just making it easier to have everyone using the latest and up-to-date version of CC.

**Comment by dgirardeau** [comment]
Ah the issue is that the CI (Github actions) is not able to compile all the plugins (far from it). Especially since some of them rely on private license keys or SDKs that can't be installed easily on a server (such as Faro, FBX, Oculus, etc.).

**Comment by jo-chemla** [comment]
Thanks for the amazingly fast feedback. If Github Actions does not build the windows releases to avoid exposing these private keys, then another way is to build the choco package after the build exes has been pushed to github releases. A choco package could be deployed - by you, another CC maintainer, or a maintainer form the chocolatey-community - and updated every time CC is updated. 

Here is the [quickstart-guide](https://docs.chocolatey.org/en-us/create/create-packages-quick-start) with [automatic packages](https://docs.chocolatey.org/en-us/create/automatic-packages) - I think the recommended way is to use [Chocolatey automatic updater module](https://github.com/majkinetor/au). We could also request some help from the chocolatey-community core members to maintain the package if you wish to.

**Comment by dgirardeau** [comment]
Yes, that would be the only viable solution... I'll try to look at that (whenever I find the time 😅 )

**Comment by rollingmoai** [comment]
> [Chocolatey](https://chocolatey.org/) is one of the most used package managers for windows 

Don't forget about [Scoop](https://scoop.sh/), [`cloudcompare`](https://scoop.sh/#/apps?q=cloudcompare&s=0&d=1&o=true) is available in the official bucket. I'm sure that most of your "Chocolatey package requests" are alternatively available on Scoop as well 😉.

**Comment by jo-chemla** [comment]
Thanks, I just discover about Scoop yesterday and indeed forgot to search for these packages there - should surely do! I use choco for ~70 package installs and updates on various machines, so there will probably be some packages not present on scoop - but using a mix of choco and scoop seems indeed the right choice if they are there already. (scoop even has an ansible integration, so again very useful)

**Comment by jo-chemla** [comment]
Note for anyone stumbling here, if the CloudCompare builds were pushed to github releases, the winget `CloudCompare.CloudCompare` package could be auto-updated on every github release. 
Otherwise, pushing a new package version with installer exe url to winget-pkgs can be done simply using eg [Komac](https://github.com/russellbanks/Komac).


Regarding the winget-releaser github-action workflow, here is a reference article based on [Winget-Releaser@v2 Github Action](https://github.com/russellbanks/Komac?tab=readme-ov-file#usage-with-github-actions-winget-releaser-). + Below is an action yml files made to keep the winget package up-to-date on every github release, via Github Actions. 

Requires:
 - You would need to create a classic Personal Access Token (PAT) with `public_repo` scope.
 - This token should be added as a repository secret by creating a [new repo secret](https://github.com/Jacalz/rymdport/settings/secrets/actions/new) with name `WINGET_ACC_TOKEN` and value the token
 - Fork [microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs) under the same account/organization as the project's repository. If you are forking winget-pkgs on a different account (e.g. bot/personal account), you can use the fork-user input to specify the username of the account where the fork is present.

```yaml
name: Publish to WinGet
on:
  release:
    types: [ released ]
jobs:
  publish:
    runs-on: windows-latest
    steps:
      - uses: vedantmgoyal9/winget-releaser@v2
        with:
          identifier: CloudCompare.CloudCompare
          token: ${{ secrets.WINGET_TOKEN }}
          installers-regex: '\w*\.exe$'
```

**Comment by dgirardeau** [comment]
One issue is that the CI (github action) is not able to compile some plugins ... But I guess it all depends what this Cholatey package is meant to be used for?

**Comment by jo-chemla** [comment]
Indeed what would be probably more useful is if the package published to winget or choco would include commonly used plugins. If the plugins missing from the github release do not have a big impact, then this automated release would probably be useful. 

Otherwise, one could also create a second winget package with all drivers from the installer that lives on the CC website, and another package which would be always up-to-date thanks to the github release, but with a smaller set of plugins.

**Comment by dgirardeau** [comment]
According to the CI script (https://github.com/CloudCompare/CloudCompare/blob/master/.github/workflows/build.yml), here are the compiled plugins for the Windows version:
            -DPLUGIN_GL_QEDL
            -DPLUGIN_GL_QSSAO
            -DPLUGIN_IO_QADDITIONAL
            -DPLUGIN_IO_QCORE
            -DPLUGIN_IO_QE57
            -DPLUGIN_IO_QPHOTOSCAN
            -DPLUGIN_IO_QLAS
            -DPLUGIN_IO_QRDB
            -DPLUGIN_STANDARD_QANIMATION
            -DPLUGIN_STANDARD_QBROOM
            -DPLUGIN_STANDARD_QCOMPASS
            -DPLUGIN_STANDARD_QCSF
            -DPLUGIN_STANDARD_QFACETS
            -DPLUGIN_STANDARD_QHOUGH_NORMALS
            -DPLUGIN_STANDARD_QHPR
            -DPLUGIN_STANDARD_QM3C2
            -DPLUGIN_STANDARD_QPCV
            -DPLUGIN_STANDARD_QPOISSON_RECON
            -DPLUGIN_STANDARD_QSRA
            -DPLUGIN_STANDARD_QRANSAC_SD
            -DPLUGIN_STANDARD_QPCL
            -DPLUGIN_STANDARD_QCLOUDLAYERS
            -DPLUGIN_STANDARD_QVOXFALL

We miss Canupo, 3DMASC, Masonry, Colorimetric Segmenter, Fit Spline, Cork, JSON RPC, MeshBoolean, MPlane, Tree ISO. And also some file formats (DotProduct, Faro, Riegl,  CSV matrix, Step).

So quite a lot 😅
